### PR TITLE
Add k8s walkthrough for outlier detection

### PR DESCRIPTION
**Pull request by fawadkhaliq**
**Issue** https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/346

**Description of changes**
- Added a k8s walkthrough for outlier detection
- Enabled for preview only. This will be changed when we move out of preview
- Created a common dir for tools / scripts that can re-used across the walkthroughs and added a traffic 
gen tool with curl and jq included

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: walkthroughs/howto-k8s-outlier-detection/README.md [hunk 1]
```diff
@@ -0,0 +1,361 @@
+
+## Overview
+
+In this walkthrough, we'll demonstrate the use of outlier detection in AWS App Mesh with EKS.
+
+Outlier detection is a form of passive health check that temporarily ejects an endpoint/host of a given service (represented by a Virtual Node) from the load balancing set when it meets failure threshold (hence considered an *outlier*). Outlier detection is support as configuration in Virtual Nodes listeners.
```
Comment (achevuru) [inline walkthroughs/howto-k8s-outlier-detection/README.md:6]: nit: support -> supported

File: walkthroughs/howto-k8s-outlier-detection/README.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+Latencies     [min, mean, 50, 90, 95, 99, max]  1.605ms, 1.995ms, 1.962ms, 2.267ms, 2.318ms, 2.441ms, 5.114ms
+Bytes In      [total, mean]                     1400, 7.00
+Bytes Out     [total, mean]                     0, 0.00
+Success       [ratio]                           100.00%
+Status Codes  [code:count]                      200:200
+Error Set:
+```
+
+The request above was made while the host `c87a6e70-c9a2-4343-a453-81808bec9d2d` was ejected and we got 100% successful responses.
+
```
Comment (achevuru) [inline walkthroughs/howto-k8s-outlier-detection/README.md:354]: Might be good to show the bad host/instance of the color app will be added back after the ejection duration.

File: walkthroughs/howto-k8s-outlier-detection/README.md (hunk 1 above)
Comment (fawadkhaliq) [reply walkthroughs/howto-k8s-outlier-detection/README.md]: thanks, fixed!

File: walkthroughs/howto-k8s-outlier-detection/README.md (hunk 2 above)
Comment (fawadkhaliq) [reply walkthroughs/howto-k8s-outlier-detection/README.md:354]: ++, added!

**Review by achevuru — APPROVED** [review]

**Pull request merged**